src/main/java/uk/gov/companieshouse/filetransferservice/service/file/transfer/S3FileStorage.java

```diff
@@ -108,18 +123,26 @@ public Optional<FileDetailsApi> getFileDetails(String fileId) {
             avCreatedOn = avTags.get(AV_TIMESTAMP_KEY);
         }
 
+        String fileName = s3File.getObjectMetadata().getUserMetaDataOf(FILENAME_METADATA_KEY);
+
```

Comment:
What if the fileName is null? Do we need any null checks here?

Reply:
If the filename is null then a null value will be returned. The only situation this would happen in is if the file was uploaded to S3 without using this service or if the metadata was somehow deleted. I think these corner cases are very rare, and even if they occur, I think returning NULL for the name would still be appropriate. The only possible default value which might be appropriate would be an empty string. This would avoid null pointer exceptions, but could lead to situations where the code is failing silently.